render/persistentvolume.go

```diff
@@ -116,3 +120,82 @@ func (v pvTocontrollerRenderer) Render(rpt report.Report) Nodes {
 	}
 	return Nodes{Nodes: nodes}
 }
+
+// SCToSPCRenderer is a Renderer which produces a renderable SPC Disk
+var SCToSPCRenderer = scTospcRenderer{}
+
+//scTospcRenderer is a Renderer to render SC & SPC.
+type scTospcRenderer struct{}
+
+//Render renders the SC & SPC nodes with adjacency.
+func (v scTospcRenderer) Render(rpt report.Report) Nodes {
+	nodes := make(report.Nodes)
+	for spcID, spcNode := range rpt.StoragePoolClaim.Nodes {
+		storagePoolClaimName, _ := spcNode.Latest.Lookup(kubernetes.Name)
+		spcname := strconv.Quote(storagePoolClaimName)
+		for scID, scNode := range rpt.StorageClass.Nodes {
+			storageClass, _ := scNode.Latest.Lookup(kubernetes.Value)
+			if strings.Contains(storageClass, ":") {
+				storagevalue := strings.Split(storageClass, ":")
+				storageClass = strings.TrimSpace(storagevalue[2])
+				if spcname == storageClass {
+					scNode.Adjacency = scNode.Adjacency.Add(spcNode.ID)
+					scNode.Children = scNode.Children.Add(spcNode)
+				}
+			}
+			nodes[scID] = scNode
+		}
+		nodes[spcID] = spcNode
+	}
+	return Nodes{Nodes: nodes}
+}
+
+// SPCToSPRenderer is a Renderer which produces a renderable Maya SP
+var SPCToSPRenderer = spcTospRenderer{}
+
+//spcTospRenderer is a Renderer to render SPC & SP nodes.
+type spcTospRenderer struct{}
+
+//Render renders the SPC & SP nodes with adjacency.
+func (v spcTospRenderer) Render(rpt report.Report) Nodes {
+	nodes := make(report.Nodes)
+	for spID, spNode := range rpt.StoragePool.Nodes {
+		storagePoolName, _ := spNode.Latest.Lookup(kubernetes.Label)
+		for spcID, spcNode := range rpt.StoragePoolClaim.Nodes {
+			storagePoolClaimName, _ := spcNode.Latest.Lookup(kubernetes.Name)
+			if storagePoolName == storagePoolClaimName {
+				spcNode.Adjacency = spcNode.Adjacency.Add(spNode.ID)
+				spcNode.Children = spcNode.Children.Add(spNode)
+			}
+			nodes[spcID] = spcNode
```

Comment:
Only one-time rendering should be done for SPC nodes, this is something I think is duplicate